wordpress-mobile/WordPress-iOS
# Autosave: Do not use autosave for drafts. Minor refactor.

Fixes #13079 
This PR addresses a situation where previewing changes made to a draft post would result in comments for that post being incorrectly turned off.  This is a quirk of the v1.1 autosave endpoint and we have opted to address the issue in the apps rather than in the API.  See p3hLNG-15Z-p2 for more background on that decision. 

I've added a check to see if the post has a draft status when autosaving. If the post or the original post is a draft, we query the API for the post status on the server.  This is an extra safety measure for the edge case where the post has been published outside of the app, but the app still has the draft.

I've also performed a minor refactor to flatten some conditionals and shorten methods. The autosave method was getting rather large with lots of nesting. 

My Objective C is a bit rusty (yay Swift!) and [block syntax](http://fuckingblocksyntax.com/) remains a right pita. Please give special attention to the block factories and typedefs when reviewing.

@shiki could I trouble you for a review of this one? 

cc @oguzkocer 

To test:
- Use the develop branch to confirm the bug by following the steps outlined in [this comment](https://github.com/wordpress-mobile/WordPress-iOS/issues/13079#issuecomment-580982420) in the issue.
- Switch to this branch and repeat the steps to confirm the bug is resolved. 
- Confirm that the autosave behavior is not altered for posts with other statuses. 

PR submission checklist:
